doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Linux/index.md

```diff
@@ -4,302 +4,232 @@ title: Linux
-
-       ~~~ shell
-       $ sudo apt-get install python-software-properties
-       $ sudo add-apt-repository ppa:ubuntu-toolchain-r/test
-       $ sudo apt-get update
-       $ sudo apt-get install gcc-4.8
-       $ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 50
-       ~~~
-
-2. Install CMake utility:
-    1. For automatic installation, execute the following commands (tested on Ubuntu 14.04):
+Before building the C++ endpoint SDK, install the following components on your machine:
 
-       ~~~ shell
-       $ sudo apt-get install cmake
-       ~~~
+1. Install g++, [CMake](https://cmake.org/download/), [Boost](http://www.boost.org/users/download/) and (optionally) [SQLite3](https://sqlite.org/download.html):
 
-    2. For manual installation, refer to the following example:
+   ```
+   sudo apt-get install g++ cmake libboost1.55-all-dev libsqlite3-0 libsqlite3-dev
+   ```
 
-       ~~~ shell
-       $ wget http://www.cmake.org/files/v3.3/cmake-3.3.0-rc2.tar.gz
-       $ tar -zxf cmake-3.3.0-rc2.tar.gz
-       $ cd cmake-3.3.0-rc2/
-       $ ./configure
-       $ sudo make install
-       ~~~
+4. Install the [AvroC++](http://avro.apache.org/docs/1.7.6/api/cpp/html/index.html) library manually:
 
-3. Install OpenSSL:
+   ```
+   wget http://archive.apache.org/dist/avro/avro-1.7.5/cpp/avro-cpp-1.7.5.tar.gz
+   tar -zxf avro-cpp-1.7.7.tar.gz
+   cd avro-cpp-1.7.5/
+   cmake -G "Unix Makefiles"
+   sudo make install
+   ```
 
-   ~~~
-   $ sudo apt-get install libssl-dev
-   ~~~
+5. Install the [Botan](http://botan.randombit.net/) 1.11 library. The stock 1.10 version is not recommended for C++11 projects,
+so the newer 1.11 version is used in Kaa C++ SDK.
+To install, proceed as follows:
 
-4. Install CUnit:
+   ```
+   wget https://github.com/randombit/botan/archive/1.11.28.tar.gz
+   tar -zxf 1.11.28.tar.gz
+   cd botan-1.11.28/
+   ./configure.py
+   sudo make install
+   sudo ln -s /usr/local/include/botan-1.11/botan /usr/local/include/botan
+   ```
 
-   ~~~
-   $ sudo apt-get install libcunit1-dev
-   ~~~
+6. Install the [SQLite](https://www.sqlite.org/index.html) library by executing the following command:
 
-## Compile C endpoint SDK
+    ```
+    sudo apt-get install libsqlite3-0 libsqlite3-dev
+    ```
+## Compiling SDK
 
-To build the C endpoint SDK, do the following:
+To build the C++ endpoint SDK, do the following:
 
-<!-- TODO: Fix this link when Administration-UI-guide#AdministrationUIguide-GeneratingSDK are updated -->
-1. [Generate ](Administration-UI-guide#AdministrationUIguide-GeneratingSDK)the C endpoint SDK in Admin UI.
-2. Download and untar the Kaa C SDK archive.
+<!-- TODO: KAA-700 -->
+1. [Generate]({{root_url}}Administration-UI-guide#AdministrationUIguide-GeneratingSDK) the C++ endpoint SDK in Admin UI.
+2. Download and untar the Kaa C++ SDK archive.
 
-   ~~~ bash
-   $ tar xfv kaa-c-ep-sdk.tar.gz
-   ~~~
+   ``` bash
+   $ tar xfv kaa-cpp-ep-sdk.tar.gz
```

Comment:
The order does not matter here. It does matter, however, if the options start with a minus.